client/visibility.js

```diff
@@ -0,0 +1,23 @@
+//heavy influence: https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API
```

Comment:
This visibility object is useful for cross browser functionality. Who would've thought each browser has their own specific hidden flag.